code/modules/mob/living/basic/jungle/seedling/seedling.dm

```diff
@@ -0,0 +1,314 @@
+	var/static/mutable_appearance/petal_active
+	///appearance when we are dead
+	var/static/mutable_appearance/petal_dead
+	///the bucket we carry
+	var/obj/item/reagent_containers/cup/held_can
+	///commands we follow
+	var/list/seedling_commands = list(
+		/datum/pet_command/idle,
+		/datum/pet_command/free,
+		/datum/pet_command/follow,
+	)
+
+/mob/living/basic/seedling/Initialize(mapload, mob/living/tamer)
+	. = ..()
+	var/datum/action/cooldown/mob_cooldown/projectile_attack/rapid_fire/seedling/seed_attack = new(src)
+	seed_attack.Grant(src)
+	ai_controller.set_blackboard_key(BB_RAPIDSEEDS_ABILITY, seed_attack)
+	var/datum/action/cooldown/mob_cooldown/solarbeam/beam_attack = new(src)
+	beam_attack.Grant(src)
+	ai_controller.set_blackboard_key(BB_SOLARBEAM_ABILITY, beam_attack)
+
+	var/petal_color = pick(possible_colors)
+
+	petal_neutral = mutable_appearance(icon, "[icon_state]_overlay")
+	petal_neutral.color = petal_color
+
+	petal_warmup = mutable_appearance(icon, "[icon_state]_charging_overlay")
+	petal_warmup.color = petal_color
+
+	petal_active = mutable_appearance(icon, "[icon_state]_fire_overlay")
+	petal_active.color = petal_color
+
+	petal_dead = mutable_appearance(icon, "[icon_state]_dead_overlay")
+	petal_dead.color = petal_color
+
+	AddElement(/datum/element/wall_smasher)
+	AddComponent(/datum/component/obeys_commands, seedling_commands)
+	RegisterSignal(src, COMSIG_HOSTILE_PRE_ATTACKINGTARGET, PROC_REF(pre_attack))
+	if(tamer)
+		befriend(tamer)
+	update_appearance()
+
+/mob/living/basic/seedling/proc/pre_attack(mob/living/puncher, atom/target)
+	SIGNAL_HANDLER
+
+	if(istype(target, /obj/machinery/hydroponics))
+		treat_hydro_tray(target)
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+	if(!held_can)
+		return
+
+	if(istype(target, /obj/structure/sink) || istype(target, /obj/structure/reagent_dispensers))
+		INVOKE_ASYNC(held_can, TYPE_PROC_REF(/obj/item, melee_attack_chain), src, target)
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+
+///seedlings can water trays, remove weeds, or remove dead plants
+/mob/living/basic/seedling/proc/treat_hydro_tray(obj/machinery/hydroponics/hydro)
+
+	if(hydro.plant_status == HYDROTRAY_PLANT_DEAD)
+		src.balloon_alert(src, "dead plant removed!")
+		hydro.set_seed(null)
+
+	if(hydro.weedlevel > 0)
+		src.balloon_alert(src, "weeds uprooted!")
+		hydro.set_weedlevel(0)
+		return
+
+	var/list/can_reagents = held_can?.reagents.reagent_list
+
+	if(!length(can_reagents))
+		return
+
+	if((locate(/datum/reagent/water) in can_reagents) && (hydro.waterlevel < hydro.maxwater))
+		INVOKE_ASYNC(held_can, TYPE_PROC_REF(/obj/item, melee_attack_chain), src, hydro)
+		return
+
+/mob/living/basic/seedling/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	. = ..()
+
+	if(!. || !proximity_flag || held_can)
+		return
+
+	if(!istype(attack_target, /obj/item/reagent_containers/cup/watering_can))
+		return
+
+	var/obj/item/can_target = attack_target
+	can_target.forceMove(src)
+	held_can = can_target
```

Comment:
Some questions, how does a seedling drop their held can

And how does a player / sentient seedling refill it

Reply:
the player can click a sink or a water tank to refill the bucket.  i didnt figure out the drop part yet if u have any good suggestions on how to make the player able to drop it